File: src/main/java/sds/offre/tache_metier/CreerUneFormule.java
package sds.offre.tache_metier;

import sds.offre.concept_metier.*;

public class CreerUneFormule {

    private IdFormuleGenerateur idFormuleGenerateur;
    private FormuleRepository formuleRepository;

    public CreerUneFormule(IdFormuleGenerateur idFormuleGenerateur, FormuleRepository formuleRepository) {
        this.idFormuleGenerateur = idFormuleGenerateur;
        this.formuleRepository = formuleRepository;
    }

    public FormuleCreee crée(Prix prix, Durée durée) {
        Formule formule = new Formule(idFormuleGenerateur.nouveauId(), durée, prix);
        formuleRepository.addOrReplace(formule);
        return FormuleCreee.de(formule);
    }
}

File: src/main/java/sds/offre/tache_metier/CreerUneFormule.kt
package sds.offre.tache_metier

import sds.offre.concept_metier.*

class CreerUneFormule(private val idFormuleGenerateur: IdFormuleGenerateur, private val formuleRepository: FormuleRepository) {

    fun crée(prix: Prix, durée: Durée): FormuleCreee {
        val formule = Formule(idFormuleGenerateur.nouveauId(), durée, prix)
        formuleRepository.addOrReplace(formule)
        return FormuleCreee(formule)
    }
}
